cmd/storage-class.go

```diff
@@ -0,0 +1,255 @@
+	// - less than N/2, if StorageClass parity is not set.
+	// - less than StorageClass Parity, if Storage class parity is set.
+	switch ssParity {
+	case 0:
+		if rrsParity >= len(globalEndpoints)/2 {
+			return errors.New("Reduced redundancy storage class parity disks should be less than " + strconv.Itoa(len(globalEndpoints)/2))
+		}
+	default:
+		if rrsParity >= ssParity {
+			return errors.New("Reduced redundancy storage class parity disks should be less than " + strconv.Itoa(ssParity))
+		}
+	}
+
+	return nil
+}
+
+// Validates the parity disks for Standard storage class
+func validateSSParity(ssParity, rrsParity int) (err error) {
+
+	// Standard storage class implies more parity than Reduced redundancy storage class. So, Standard storage parity disks should be
+	// - greater than or equal to 2, if RRS parity is not set.
+	// - greater than RRS Parity, if RRS parity is set.
+	switch rrsParity {
+	case 0:
+		if ssParity < minimumParityDisks {
+			return errors.New("Standard storage class parity disks should be greater than or equal to " + strconv.Itoa(minimumParityDisks))
+		}
+	default:
+		if ssParity <= rrsParity {
+			return errors.New("Standard storage class parity disks should be greater than " + strconv.Itoa(rrsParity) + " parity disks")
+		}
+	}
+
+	// Standard storage class parity should be less than or equal to N/2
+	if ssParity > len(globalEndpoints)/2 {
+		return errors.New("Standard storage class parity disks should be less than or equal to " + strconv.Itoa(len(globalEndpoints)/2))
+	}
+
+	return nil
+}
+
+// Returns the data and parity drive count based on storage class
+// If storage class is set using the env vars MINIO_STORAGE_CLASS_RRS and MINIO_STORAGE_CLASS_STANDARD
+// -- corresponding values are returned
+// If storage class is not set using environment variables, default values are returned
+// -- Default for Reduced Redundancy Storage class is, parity = 2 and data = N-Parity
+// -- Default for Standard Storage class is, parity = N/2, data = N/2
+// If storage class is not present in metadata, default value is data = N/2, parity = N/2
+func getDrivesCount(sc string, disks []StorageAPI) (data, parity int) {
+	totalDisks := len(disks)
+	switch sc {
+	case reducedRedundancyStorageClass:
+		if globalRRStorageClass.Parity != 0 {
+			// set the parity if available
+			parity = globalRRStorageClass.Parity
+			data = totalDisks - parity
+		} else {
+			// else fall back to default values
+			parity = defaultRRSParity
+			data = totalDisks - parity
+		}
+	case standardStorageClass:
+		if globalStandardStorageClass.Parity != 0 {
+			// set the parity if available
+			parity = globalStandardStorageClass.Parity
+			data = totalDisks - parity
+		} else {
+			// else fall back to default values
+			parity = totalDisks / 2
+			data = totalDisks / 2
+		}
+	default:
+		parity = totalDisks / 2
+		data = totalDisks / 2
+	}
+	return data, parity
```

Comment:
you can compute `data` here once because for all cases it must necessarily be `totalDisks - parity` (this would also mimize risk of intoducing errors when new class would be introduced with somebody making an error in computing data for a particular case

Reply:
good idea! done.